test/fakeintake/server/server.go

```diff
@@ -87,11 +87,15 @@ func NewServer(options ...func(*Server)) *Server {
 
 	registry := prometheus.NewRegistry()
 
+	storeMetrics := fi.store.GetMetrics()
```

Comment:
Is it really useful? Do we have some case where a newly instantiated serverstore already contains metrics?

Reply:
I think the store must register its metrics and the metrics will depend on the store implementation. For example on SQL we might want SQL metrics

Reply:
Oh I missed the line `fi.store.NbPayloads` in the diff. Looks good to me then, maybe we could just have the naming a bit more explicit to indicate that we actually retrieve the number of payload in the store as a Prometheus metric. Otherwise it can be a bit misleading since the store can be used to store metrics sent from the agent

Reply:
Do you have any suggestion on a better name ? `GetMetrics()` can return any metric we define for the given store. For example on SQL it should return latencies and SQL metrics